### ISSUE TITLE
Typescript compile errors (WebAssembly not defined and others)

### ISSUE THREAD

**Issue by EyeOfPython** [bitauth/bitcoin-ts#30]
* **I'm submitting a ...**
[x] bug report

* **Summary**

When using typescript@3.4.5, I get a `Cannot find namespace 'WebAssembly'` error. When I switch to typescript@3.5.3, I get multiple `Cannot find name 'SharedArrayBuffer'.` and `Cannot find name 'BigInt'.`

* **Other information**

My tsconfig.json looks like this:

```json5
{
  "compilerOptions": {
    "sourceMap": true,
    "module": "commonjs",
    "moduleResolution": "node",
    "strictNullChecks": true,
    "noImplicitAny": true,
    "esModuleInterop": true,
    "declaration": true,
    "target": "es2015",
    "outDir": "dist",
  },
}
```

The error for typescript@3.4.5:

```
node_modules/bitcoin-ts/build/main/lib/bin/secp256k1/secp256k1-wasm-types.d.ts:80:24 - error TS2503: Cannot find namespace 'WebAssembly'.

80     readonly instance: WebAssembly.Instance;
                          ~~~~~~~~~~~


Found 1 error.
```

The error for typescript@3.5.3:

```
node_modules/bitcoin-ts/build/main/lib/auth/instruction-sets/common/types.d.ts:5:51 - error TS2304: Cannot find name 'BigInt'.

5 export declare const isScriptNumberError: (value: BigInt | ScriptNumberError) => value is ScriptNumberError;
                                                    ~~~~~~

node_modules/bitcoin-ts/build/main/lib/bin/secp256k1/secp256k1-wasm.d.ts:14:70 - error TS2304: Cannot find name 'SharedArrayBuffer'.

14 export declare const getEmbeddedSecp256k1Binary: () => ArrayBuffer | SharedArrayBuffer;
                                                                        ~~~~~~~~~~~~~~~~~

node_modules/bitcoin-ts/build/main/lib/crypto/ripemd160.d.ts:56:70 - error TS2304: Cannot find name 'SharedArrayBuffer'.

56 export declare const getEmbeddedRipemd160Binary: () => ArrayBuffer | SharedArrayBuffer;
                                                                        ~~~~~~~~~~~~~~~~~

node_modules/bitcoin-ts/build/main/lib/crypto/sha256.d.ts:56:67 - error TS2304: Cannot find name 'SharedArrayBuffer'.

56 export declare const getEmbeddedSha256Binary: () => ArrayBuffer | SharedArrayBuffer;
                                                                     ~~~~~~~~~~~~~~~~~

node_modules/bitcoin-ts/build/main/lib/crypto/sha512.d.ts:56:67 - error TS2304: Cannot find name 'SharedArrayBuffer'.

56 export declare const getEmbeddedSha512Binary: () => ArrayBuffer | SharedArrayBuffer;
                                                                     ~~~~~~~~~~~~~~~~~


Found 5 errors.
```

**Comment by EyeOfPython** [comment]
Adding 

```json5
"lib": [
  "es2017",
  "esnext.bigint",
  "dom",
]
```

to tsconfig.json solves this.

It would be good if the requirement for a recent typescript version and the `lib` entries were specified somewhere. Do you mind adding a paragraph for that in the README to avoid others getting bamboozled?

**Comment by bitjson** [comment]
@EyeOfPython thanks for opening an issue! Great idea, I'll add a note to the readme about it. 

I stumbled on [the same issue](https://github.com/microsoft/web-build-tools/issues/1370#issuecomment-510337327) a little while ago –  the `WebAssembly` namespace was added to `lib.dom.d.ts` in https://github.com/microsoft/TypeScript/commit/1e7482ad05034fd92254bf4a5119a895f64e89e3, which didn't get released until `v3.5.1`. Before that, this library shipped with the DefinitelyTyped types, so this only became a problem in the latest release.
